os/tools/mkbinheader.py

```diff
@@ -116,206 +177,197 @@ def get_static_ram_size(bin_type):
+            loading_priority = LOADING_HIGH
+        elif loading_priority == 'M' or loading_priority == 'MID' :
+            loading_priority = LOADING_MID
+        elif loading_priority == 'L' or loading_priority == 'LOW' :
+            loading_priority = LOADING_LOW
+        else : #Not supported.
+            print("Error : Not supported Binary loading priority")
+            sys.exit(1)
+
+        static_ram_size = get_static_ram_size(bin_type)
+        cfg_path = CONFIG_DIR + '/.config'
+        if check_optimize_config(cfg_path) == True:
+            binary_ram_size = int(dynamic_ram_size)
+        else:
+            binary_ram_size = int(static_ram_size) + int(dynamic_ram_size)
+            binary_ram_size = roundup_power_two(binary_ram_size)
+
+        if float(kernel_ver) <= 0.0 :
+            kernel_ver = 3.0
+        else :
+            kernel_ver = float(kernel_ver)
+
+        # based on comp_enabled, check if we need to compress binary.
+        # If yes, assign to bin_comp value for compression algorithm to use.
+        # Else, assign 0 to bin_comp to represent no compression
+        if 0 < int(comp_enabled) <= COMP_MAX :
+            bin_comp = int(comp_enabled)
+        else :
+            bin_comp = 0
+
+        # Compress data according to Compression Algorithm represented by bin_comp
+        # Run mkcompressimg tool with provided options. Read output compressed file into data.
+        if bin_comp > COMP_NONE :
+            fp_tmp = open("tmp", 'wb+')
+            fp_tmp.write(data)
+            fp_tmp.close()
+            os.system(mkbinheader_path + '/compression/mkcompressimg ' + comp_blk_size + ' ' + comp_enabled + ' tmp' + ' tmp_comp')
+            fp_tmp = open("tmp_comp", 'rb')
+            data = fp_tmp.read()
+            file_size = fp_tmp.tell()
+            fp_tmp.close()
+            os.system('rm tmp tmp_comp')
+
+        fp = open(file_path, 'wb')
+
+        fp.write(struct.pack('H', header_size))
+        fp.write(struct.pack('B', bin_type))
+        fp.write(struct.pack('B', bin_comp))
+        fp.write(struct.pack('B', main_priority))
+        fp.write(struct.pack('B', loading_priority))
+        fp.write(struct.pack('I', file_size))
+        fp.write('{:{}{}.{}}'.format(binary_name, '<', SIZE_OF_BINNAME, SIZE_OF_BINNAME - 1).replace(' ','\0'))
+        fp.write(struct.pack('I', int(binary_ver)))
+        fp.write(struct.pack('I', binary_ram_size))
+        fp.write(struct.pack('I', int(main_stack_size)))
+        fp.write(struct.pack('f', kernel_ver))
+        fp.write(data)
+
+        fp.close()
 
-    fp.write(struct.pack('I', addr))
-    fp.write(data)
+############################################################################
+#
+# Generate headers for binary types
+#
+############################################################################
+file_path  = sys.argv[1]
+binary_type = sys.argv[2]
+if binary_type == 'kernel' :
+    make_kernel_binary()
```

Comment:
This is not making binary.
How about changing `make_kernel_bin_header()` and `make_user_bin_header()`?
This script name is also mkbinheader.